moncli/types.py

```diff
@@ -211,4 +214,78 @@ def validate_timezone(self, value):
         try:
             pytz.timezone(value)
         except (UnknownTimeZoneError):
-            raise ValidationError('Unknown time zone "{}".'.format(value))
\ No newline at end of file
+            raise ValidationError('Unknown time zone "{}".'.format(value))
+
+class StatusType(MondayType):
+    native_type = str
+    allow_casts = (int, str)
+    null_value = {}
+    def __init__(self, id: str = None, title: str = None,data_mapping:dict =None ,  *args, **kwargs):
+        if data_mapping:
+            values= [value for value in data_mapping.values()]        
+            self.native_type = values[0].__class__
+            self.choices = values
+        self._data_mapping =  data_mapping 
+        super().__init__(id=id, title=title, *args, **kwargs)
+    
+    def _process(self, value):
+        labels = self.metadata['labels']
+        if self.native_type == str:
+            try:
+                if isinstance(int(value),int):
+                    if value in labels.values():
+                        return value
+            except ValueError:
+                    raise ConversionError('Cannot find status label with index "{}".'.format(value))
+            if isinstance(value,int):
```

Comment:
This is wrong.  At this point, we should assume that the value is a str that cannot be converted to an int, so it is assumed to be a label.  Get rid of this check.  Instead raise the Conversion error if the value is not in labels.values().